subsys/net/ip/ipv6.c

```diff
@@ -273,7 +273,10 @@ static enum net_verdict ipv6_route_packet(struct net_pkt *pkt,
 {
 	struct net_route_entry *route;
 	struct in6_addr *nexthop;
+	struct net_if *iface;
+	struct net_linkaddr *lladdr_src;
 	bool found;
+	int ret;
```

Comment:
move all these new variables down at line 333

Reply:
Or should i implement a new function in route.c containing the if-body?

Reply:
sure why not

Reply:
done